src/lava/proc/graded/models.py

```diff
@@ -0,0 +1,148 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+import numpy as np
+import typing as ty
+
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+from lava.magma.core.model.py.ports import PyInPort, PyOutPort
+from lava.magma.core.model.py.type import LavaPyType
+from lava.magma.core.resources import CPU
+from lava.magma.core.decorator import implements, requires, tag
+from lava.magma.core.model.py.model import PyLoihiProcessModel
+
+from lava.proc.graded.process import GradedVec, NormVecDelay, InvSqrt
+
+
+class AbstractGradedVecModel(PyLoihiProcessModel):
+    a_in = None
```

Comment:
Are this supposed to be Vars? They should be LavaPyTypes(..) I guess?

In PyGradedVecModelFixed one does not need to specify LavaPyTypes again then, only specific ones to PyGradedVecModelFixed if there would be any.

Reply:
This is the convention other models are using.